source/object-storage/advanced.rst

```diff
@@ -228,39 +228,67 @@ Temporary URL
 
 This is a means by which a temporary URL can be generated, to allow
 unauthenticated access to a Swift object at a given path. The
-access is via the given HTTP method (e.g. GET, PUT) and is valid
-for the number of seconds specified when the URL is created.
+access is via the given HTTP method (e.g. GET, PUT) and is valid until an
+expiry time is reached.
 
-The expiry time can be expressed as valid for the given number of seconds from
-now or if the optional --absolute argument is provided, seconds is instead
-interpreted as a Unix timestamp at which the URL should expire.
+There are several items of information needed to create a temporary URL:
+
+- The temporary URL secret key for your account, which is used to encode details in the temporary URL.
+- The URL path to the object you want to share.
+- When you want to the temporary URL to expire.
+- The HTTP method allowed to access the object, such as "GET" or "PUT"
 
-The syntax for the tempurl creation command is:
+The swift command line tool
+===========================
 
-``$ swift tempurl [command-option] [method] [seconds] [path] [key]``
+The swift command line tool is the recommended way to generate a temporary URL.
+The syntax for the temp-url creation command is:
+
+``$ swift tempurl [command-option] [method] [time] [path] [key]``
 
 This generates a temporary URL allowing unauthenticated access to the Swift
 object at the given path.
 
+The expiry time can be expressed as valid for the given number of seconds from
+now or if the optional --absolute argument is provided, seconds is instead
+interpreted as a Unix timestamp at which the URL should expire.  The expiry
+time can also be given as a date time string in the format 'YYYY-mm-dd',
+'YYYY-mm-ddTHH:MM:ss' or 'YYYY-mm-ddTHH:MM:ssZ' (the last is in UTC).
+
 For example:
 
 .. code-block:: bash
 
-  $ swift tempurl GET $(date -d "Jan 1 2017" +%s) /v1/AUTH_foo/bar_container/quux.md my_secret_tempurl_key --absolute
+  $ swift tempurl GET $(date -d "Jan 1 2023" +%s) /v1/AUTH_foo/bar_container/quux.md my_secret_tempurl_key --absolute
 
-- sets the expiry using the absolute method to be Jan 1 2017
+- AUTH_foo is the object store account id
+- sets the expiry using the absolute method to be Jan 1 2023
 - for the object : quux.md
 - in the nested container structure : bar_container/quux.md
 - with key : my_secret_tempurl_key
 
+Or by passing the expiry date:
+
+.. code-block:: bash
+
+  $ swift tempurl GET 2023-01-01 /v1/AUTH_foo/bar_container/quux.md my_secret_tempurl_key
+
 
 Creating temporary URLs in the Catalyst Cloud
 =============================================
 
-Currently, the only method available for the creation of temporary URLs is
-through the use of the command line tools.
+Currently, there two methods available for the creation of temporary URLs:
+through the use of the command line tools or by generating the temporary URL
+programmatically.
```

Comment:
perhaps "via our API" instead? Since you can run the command line tool from a program.

Reply:
The swift tool does hit our API to generate the temp URL so yeah it's hard to differentiate.